### ISSUE TITLE
I don't have an issue, I would like to ask question.

### ISSUE THREAD

**Issue by novanmartejares** [gneiru/kd#88]
Good day.

I really like your repository like a lot but
I am having a confusion regarding .env.

Particularly in database_url = 

Can you further explain to me about what to put or any methods.
Thank you for your response.

**Comment by gneiru** [comment]
> Can you further explain to me about what to put or any methods.

As of now, the repository is using mysql from planetscale. You can get your connection uri at https://planetscale.com, sign up with credit card then connect (select nodejs from dropdown menu).

P.S. They are about to drop free tier at April 8th. Speaking of which, I am planning to migrate this repo to use neonDB PostgreSQL since it's free and I don't afford planetscale's scaler plan.

**Comment by novanmartejares** [comment]
is there any changes of codes if you migrate it?

**Comment by gneiru** [comment]
> is there any changes of codes if you migrate it?

yes

**Comment by gneiru** [comment]
there will be a lot. not only because of migrating from planetscale to neon but also because of migrating from mysql to postgresql.

**Comment by novanmartejares** [comment]
Ok, thanks for the clarification. i will wait for your update then.

**Comment by novanmartejares** [comment]
its so nice.. thanks bro.. its a big help..let me try it

**Comment by novanmartejares** [comment]
hi @gneiru i am having this error when i try npm install ❌ Invalid environment variables: { DATABASE_URL: [ 'Invalid input: must include "neondb"' ] }
C:\Users\USER\Desktop\kd-master\node_modules\@t3-oss\env-core\dist\index.js:29
        throw new Error("Invalid environment variables");

i have already put all the requirements in .env.

**Comment by gneiru** [comment]
> hi @gneiru i am having this error when i try npm install ❌ Invalid environment variables: { DATABASE_URL: [ 'Invalid input: must include "neondb"' ] }
> 
> C:\Users\USER\Desktop\kd-master\node_modules\@t3-oss\env-core\dist\index.js:29
> 
>         throw new Error("Invalid environment variables");
> 
> 
> 
> i have already put all the requirements in .env.

What is your database name? I thought "neondb" is the default name so it should work.

**Comment by novanmartejares** [comment]
i think it has something to do with this code? "DATABASE_URL: z.string().url().includes("neondb")"
i guess :)

**Comment by novanmartejares** [comment]
i already figure it out. i need to change the neondb to my databasename

**Comment by gneiru** [comment]
> i already figure it out. i need to change the neondb to my databasename

Nice catch

**Comment by novanmartejares** [comment]
![kdrama](https://github.com/gneiru/kd/assets/16984585/959b1404-6c09-4535-bc12-15ff81e9b6be)

but im having this issue again :D

**Comment by gneiru** [comment]
I think your api requests are failing. Please check the enpoints in `src/lib/dramacool`. It should return JSON with matching types

**Comment by novanmartejares** [comment]
i use consumet api and host my own. the api seems to be ok in the hosting
side.

**Comment by novanmartejares** [comment]
maybe because of the type. i dont know it

**Comment by gneiru** [comment]
Try checking the endpoints, my API_URL ends with `/api`

**Comment by novanmartejares** [comment]
this is the consumet api i have hosted. and this is the one i put in the
env.

https://anime-api-ashen.vercel.app

**Comment by gneiru** [comment]
What's the endpoimt of getting trending, recent etc?

**Comment by novanmartejares** [comment]
i just put dramacool in the url like this

https://anime-api-ashen.vercel.app/movies/dramacool/recent

and it returns like this:

{"currentPage":1,"totalPages":1,"hasNextPage":false,"results":[{"id":"drama-detail/get-old-recently","title":"Get
Old Recently (2021)","url":"https://dramacool.com.pa/drama-detail/get-old-recently","image":"https://asianimg.pro/cover/get-old-recently.png"}]}

does it mean its ok now?

**Comment by gneiru** [comment]
You should put https://anime-api-ashen.vercel.app/movies/dramacool in your .env file. However, the response schema doesnt match with the app's

**Comment by novanmartejares** [comment]
i see. this is what in my mind also. let me try again.
Thanks for the help. your such a big help.

**Comment by novanmartejares** [comment]
everthing looks fine until i got this error. this is when i click button going to home page
![eror](https://github.com/gneiru/kd/assets/16984585/cedd138c-d1b9-4c4b-aab9-138d39636342)

**Comment by gneiru** [comment]
Add this code to your dramacool parser methods

`featured = async () => {
    try {
      const url = new URL(`${this.baseUrl}`);
      const res = await this.client.get(url.toString());

      const $ = load(res.data);

      const featured: IMovieEpisode[] = [];

      $("#layerslider .ls-slide").each((i, el) => {
        featured.push({
          id: $(el).find("a").attr("href")?.slice(1)!,
          title: $(el).find("img").attr("title")!,
          image: $(el).find("img").attr("src"),
        });
      });
      return featured;
    } catch (err) {
      throw new Error("Something went wrong. Please try again later.");
    }
  };`

Then add a route to call this method, in my case it is `/featured`

**Comment by novanmartejares** [comment]
im sorry for not getting it. im so noob right now. but should  i put it here?

![dramacool](https://github.com/gneiru/kd/assets/16984585/c09019ac-6237-4f5a-9d2f-74c0c16ea508)

**Comment by gneiru** [comment]
No, in your self hosted code. If you want, I'll make a PR on your repo of the self hosted consumet API

**Comment by novanmartejares** [comment]
Yes, of course you can!

**Comment by gneiru** [comment]
Which repo? Is it public? If not, could you please add me as collaborator.

**Comment by novanmartejares** [comment]
https://github.com/novanmartejares/anime_api

**Comment by gneiru** [comment]
Can't access it for some reason

**Comment by novanmartejares** [comment]
i think its private

**Comment by novanmartejares** [comment]
i have added you as collaborator

**Comment by novanmartejares** [comment]
animeapi-two.vercel.app

this is the api.. not the other one

**Comment by gneiru** [comment]
I can't add the code unfortunately. Your API is using the API wrapper of consumet.ts, while mine is a fork of consumet.ts that has API routes in the same codebase. All I can offer is give you my API_URL, which is https://v1.api.ani.rohi.dev/api/dramacool, my apologies.

**Comment by novanmartejares** [comment]
it's now working. it displays the homepage and popular. I can also search now. but there's an error occurred.
Upon clicking the image box to play the video it will display like this
 
![Screenshot 2024-04-01 220749](https://github.com/gneiru/kd/assets/16984585/853d4029-83f7-4307-8505-d67667f47097)

**Comment by gneiru** [comment]
You need to run `pnpm db push` on your terminal

**Comment by novanmartejares** [comment]
wow .. its working now.. :)

**Comment by novanmartejares** [comment]
i use this command pnpm drizzle-kit push:pg

**Comment by skgupta507** [comment]
hello 
@gneiru while deploying it to vercel or netlify i got a error message .
<html>
<body>

build: Error occurred prerendering page "/sitemap.xml". Read more: https://nextjs.org/docs/messages/prerender-error
--
11:49:18.034 | build:
11:49:18.034 | build: error: relation "kd_episode" does not exist
11:49:18.034 | build:     at /vercel/path0/.next/server/app/sitemap.xml/route.js:9:29101
11:49:18.034 | build:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
11:49:18.034 | build:     at async n_.execute (/vercel/path0/.next/server/app/sitemap.xml/route.js:33:28439)
11:49:18.034 | build:     at async at (/vercel/path0/.next/server/app/sitemap.xml/route.js:33:38298)
11:49:18.034 | build:     at async aa (/vercel/path0/.next/server/app/sitemap.xml/route.js:33:39203)
11:49:18.034 | build:     at async /vercel/path0/node_modules/.pnpm/next@14.1.4_@babel+core@7.24.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:42484
11:49:18.034 | build:     at async eI.execute (/vercel/path0/node_modules/.pnpm/next@14.1.4_@babel+core@7.24.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:32486)
11:49:18.035 | build:     at async eI.handle (/vercel/path0/node_modules/.pnpm/next@14.1.4_@babel+core@7.24.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:43737)
11:49:18.035 | build:     at async exportAppRoute (/vercel/path0/node_modules/.pnpm/next@14.1.4_@babel+core@7.24.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/routes/app-route.js:77:26)
11:49:18.035 | build:     at async exportPageImpl (/vercel/path0/node_modules/.pnpm/next@14.1.4_@babel+core@7.24.0_react-dom@18.2.0_react@18.2.0/node_modules/next/dist/export/worker.js:175:20)
11:49:18.035 | build:  build:  ✓ Generating static pages (14/14)
11:49:18.041 | build:
11:49:18.041 | build: > Export encountered errors on following paths:
11:49:18.041 | build: 	/sitemap.xml/route: /sitemap.xml
11:49:18.109 | build:  ELIFECYCLE  Command failed with exit code 1.
11:49:18.122 | build: ERROR: command finished with error: command (/vercel/path0/) /pnpm8/node_modules/.bin/pnpm run build exited (1)
11:49:18.122 | build: command (/vercel/path0/) /pnpm8/node_modules/.bin/pnpm run build exited (1)
11:49:18.123 |  
11:49:18.123 | Tasks:    0 successful, 1 total
11:49:18.123 | Cached:    0 cached, 1 total
11:49:18.123 | Time:    44.779s
11:49:18.123 | Summary:    /vercel/path0/.turbo/runs/2ecl4rkuLRFdTlXi1yhBB2sjuEX.json
11:49:18.123 | Failed:    //#build
11:49:18.123 |  
11:49:18.123 | ERROR  run failed: command  exited (1)
11:49:18.182 | Error: Command "turbo run build" exited with 1


</body>
</html>

i recieved this error and i have no idea how to fix this can u help me with this

**Comment by gneiru** [comment]
> i use this command pnpm drizzle-kit push:pg

This command also works

**Comment by skgupta507** [comment]
> > i use this command pnpm drizzle-kit push:pg
> 
> This command also works

i got this error for this command
<html>
<body>

ERR_PNPM_RECURSIVE_EXEC_FIRST_FAIL  Command "drizzle-kit" not found
--
12:35:25.167 | Error: Command "pnpm drizzle-kit push:pg" exited with 1


</body>
</html>

**Comment by gneiru** [comment]
Try `pnpm i` first then `pnpm db:push`

**Comment by gneiru** [comment]
> @gneiru 
> 
> bro i tried but it stll says drizzle-kit not found
> 
> 
> 
> <html>
> 
> <body>
> 
> 
> 
> sh: line 1: drizzle-kit: command not found
> 
> --
> 
> 12:59:37.167 | ELIFECYCLE  Command failed.
> 
> 
> 
> 12:59:37.174 | WARN   Local package.json exists, but node_modules missing, did you mean to install?
> 
> 
> 
> 12:59:37.196 | Error: Command "pnpm db:push" exited with 1
> 
> 
> 
> 
> 
> </body>
> 
> </html>

You dont have node modules so it will not work. You have to run pnpm install first, then wait for node modules to appear in your root directory. Finally, pnpm db:push will definitely not throw error.

**Comment by skgupta507** [comment]
> > @gneiru
> > bro i tried but it stll says drizzle-kit not found
> > 
> > 
> > 
> > sh: line 1: drizzle-kit: command not found
> > --
> > 12:59:37.167 | ELIFECYCLE  Command failed.
> > 12:59:37.174 | WARN   Local package.json exists, but node_modules missing, did you mean to install?
> > 12:59:37.196 | Error: Command "pnpm db:push" exited with 1
> 
> You dont have node modules so it will not work. You have to run `pnpm install` first, then wait for node modules to appear in your root directory. Finally, `pnpm db:push` will definitely not throw error.

Hi @gneiru thanks to ur help i was able to deploy the site sucessfully.

but i have another help to ask with authentication when  i do the signin with google i get the eror 500:internal server error.
can u help me how to fix this.  this is my Redirect_uri
https://kdramax.vercel.app/api/auth/callback/google

**Comment by gneiru** [comment]
Check your project's logs in vercel, it might help us debug the error

**Comment by skgupta507** [comment]
> Check your project's logs in vercel, it might help us debug the error

here`s the log file
[logs_result.json](https://github.com/gneiru/kd/files/14969826/logs_result.json)

it says something about environments variables being invalid.

**Comment by gneiru** [comment]
Ah. Then, make sure you have env values for google client id and secret. Once you're done, hit redeploy on your latest production deployment. 

Regarding edge invocation failing, what I can help atleast is to remove the line of code which set the runtime to edge in your `api/auth/[...next-auth]/route.ts`. If it still fails dynamically, let me know but edge in my case works fine. Vercel's edge function in your vercel workspace might be a factor too!

**Comment by skgupta507** [comment]
> Ah. Then, make sure you have env values for google client id and secret. Once you're done, hit redeploy on your latest production deployment.
> 
> Regarding edge invocation failing, what I can help atleast is to remove the line of code which set the runtime to edge in your `api/auth/[...next-auth]/route.ts`. If it still fails dynamically, let me know but edge in my case works fine. Vercel's edge function in your vercel workspace might be a factor too!

i have already added all env variables before deploying the app but
the error accurs after the app is authorized
